### PR TITLE
Named Part Fields are not indexed

### LINKED ISSUE #9444: Content fields on named parts are not indexed in SQL
### Describe the bug

Content field values are not indexed if those are attached to a content part used as a named part on a content item. Fields attached directly to the content type, however, are indexed as expected.

### To Reproduce

Prerequisites: Make sure SQL field indexing feature is enabled.

1. Go to Admin UI > Content > Content Parts
2. Create a Content Part and edit it
    - Set Reusable to true
    - Add some field (e.g., a TextField)
    - Save
3. Go to Admin UI > Content > Content Types
4. Create a Content Type and edit it
    - Add a named part using the content part you've created previously.
    - Add a content field to the content type directly for observing proper indexing for this field
    - Save
5. Create a content item using the type you've previously created
    - Set the two field values you've attached: the one from the named part and the other one attached directly to the type
    - Save
6. Open SSMS / DB Browser for SQLite depending on the database type.
    - Open the related index table (e.g., TextFieldIndex)

### Actual result
Observe the field value on the named part is not indexed, however, the one attached to the type directly is properly indexed.

### Expected behavior
The field attached to the named part should be indexed. The ContentPart value should be the name of the named part.

### PR DESCRIPTION
**Pull request by lampersky**
fixes #9444

@barthamark I faced the same issue, and fixed it in this PR.

/cc @sebastienros or @jtkech

### CODE REVIEW

**Comment by lampersky** [comment]
@sebastienros Could you please take a look?

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Extensions/ContentPartFieldDefinitionExtensions.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static class ContentPartFieldDefinitionExtensions
         ContentItem contentItem)
         where TField : ContentField
     {
-        if (((JsonObject)contentItem.Content)[fieldDefinition.PartDefinition.Name] is not JsonObject jPart ||
+        if (((JsonObject)contentItem.Content)[fieldDefinition.ContentTypePartDefinition.Name] is not JsonObject jPart ||
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Extensions/ContentPartFieldDefinitionExtensions.cs:16]: For those wondering: `fieldDefinition.PartDefinition.Name` here will always be the technical name of the content part itself, as also listed under /Admin/ContentTypes/ListParts. `fieldDefinition.ContentTypePartDefinition.Name` will be the same for non-reusable parts, however, it'll be the instance name for reusable named parts.

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/SQL/BooleanFieldIndexProvider.cs [hunk 2]
```diff
@@ -77,7 +77,7 @@ public override void Describe(DescribeContext<ContentItem> context)
                                 ContentItemId = contentItem.ContentItemId,
                                 ContentItemVersionId = contentItem.ContentItemVersionId,
                                 ContentType = contentItem.ContentType,
-                                ContentPart = pair.Definition.PartDefinition.Name,
+                                ContentPart = pair.Definition.ContentTypePartDefinition.Name,
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/SQL/BooleanFieldIndexProvider.cs:80]: This is not unique among content types, i.e. you can attach a named part with the name "Example" to more than one content type, even different named parts. I don't think this is an issue, but worth keeping in mind. Please check out if this causes any issues, including with SQL indices (it doesn't seem so; unless you created custom unique indices on just that column).

**Review by Piedone — APPROVED** [review]
I sincerely apologize for us taking so much time here. I checked out the PR, going over old ones, and this would be quite useful.

I checked this out and it seems to be correct. However, this might have broader implications than I can grasp, so I asked this to be covered during the weekly triage meeting too.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**